// webflow-api/dist/serialization/types/ProductFieldData.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductFieldData = void 0;
const core = __importStar(require("../../core"));
const SkuPropertyList_1 = require("./SkuPropertyList");
const ProductFieldDataTaxCategory_1 = require("./ProductFieldDataTaxCategory");
const ProductFieldDataEcProductType_1 = require("./ProductFieldDataEcProductType");
exports.ProductFieldData = core.serialization.object({
    name: core.serialization.string().optional(),
    slug: core.serialization.string().optional(),
    description: core.serialization.string().optional(),
    shippable: core.serialization.boolean().optional(),
    skuProperties: core.serialization.property("sku-properties", core.serialization.list(SkuPropertyList_1.SkuPropertyList).optional()),
    categories: core.serialization.list(core.serialization.string()).optional(),
    taxCategory: core.serialization.property("tax-category", ProductFieldDataTaxCategory_1.ProductFieldDataTaxCategory.optional()),
    defaultSku: core.serialization.property("default-sku", core.serialization.string().optional()),
    ecProductType: core.serialization.property("ec-product-type", ProductFieldDataEcProductType_1.ProductFieldDataEcProductType.optional()),
});
