// webflow-api/serialization/types/Page.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Page = void 0;
const core = __importStar(require("../../core"));
const PageSeo_1 = require("./PageSeo");
const PageOpenGraph_1 = require("./PageOpenGraph");
exports.Page = core.serialization.object({
    id: core.serialization.string(),
    siteId: core.serialization.string().optional(),
    title: core.serialization.string().optional(),
    slug: core.serialization.string().optional(),
    parentId: core.serialization.string().optional(),
    collectionId: core.serialization.string().optional(),
    createdOn: core.serialization.date().optional(),
    lastUpdated: core.serialization.date().optional(),
    archived: core.serialization.boolean().optional(),
    draft: core.serialization.boolean().optional(),
    canBranch: core.serialization.boolean().optional(),
    isMembersOnly: core.serialization.boolean().optional(),
    seo: PageSeo_1.PageSeo.optional(),
    openGraph: PageOpenGraph_1.PageOpenGraph.optional(),
    localeId: core.serialization.string().optional(),
    publishedPath: core.serialization.string().optional(),
});
